docs/testing/integration-tests.md

```diff
@@ -0,0 +1,200 @@
+
+Notice that the layout within `test_increment()`. Every `.call()` obtains its required gas from the account performing it. Unlike the unit test, there is no mocking being performed before the call as the context is provided by the environment initialized during `main()`. Every call interacts with this environment to either fetch or change state. 
+
+:::info
+**Pitfall**: you must compile your contract before running simulation tests. Because workspaces tests use the `.wasm` files to deploy the contracts to the network. If changes are made to the smart contract code, the smart contract wasm should be rebuilt before running these tests again.
+:::
+
+## Helpful Snippets
+
+### Create an Account
+
+```rust reference
+https://github.com/near-examples/rust-counter/blob/6a7af5a32c630e0298c09c24eab87267746552b2/integration-tests/rs/src/tests.rs#L16-L21
+```
+
+### Create Helper Functions
+
+```rust reference
+https://github.com/near-examples/nft-tutorial/blob/7fb267b83899d1f65f1bceb71804430fab62c7a7/integration-tests/rs/src/helpers.rs#L148-L161
+```
+
+### Spooning - Pulling Existing State and Contracts from Mainnet/Testnet
+
+This example showcases spooning state from a testnet contract into our local sandbox environment:
+
+```rust reference
+https://github.com/near/workspaces-rs/blob/c14fe2aa6cdf586028b2993c6a28240f78484d3e/examples/src/spooning.rs#L64-L122
+```
+
+For a full example, see the [examples/src/spooning.rs](https://github.com/near/workspaces-rs/blob/main/examples/src/spooning.rs) example.
+
+### Fast Forwarding - Fast Forward to a Future Block
+
+`workspaces` testing offers support for forwarding the state of the blockchain to the future. This means contracts which require time sensitive data do not need to sit and wait the same amount of time for blocks on the sandbox to be produced. We can simply just call `worker.fast_forward` to get us further in time:
+
+```rust reference
+https://github.com/near/workspaces-rs/blob/c14fe2aa6cdf586028b2993c6a28240f78484d3e/examples/src/fast_forward.rs#L12-L44
+```
+
+For a full example, take a look at [examples/src/fast_forward.rs](https://github.com/near/workspaces-rs/blob/main/examples/src/fast_forward.rs).
+
+### Handle Errors
+
+```rust reference
+https://github.com/near-examples/FT/blob/98b85297a270cbcb8ef3901c29c17701e1cab698/integration-tests/rs/src/tests.rs#L199-L225
```

Comment:
I am not sure if panicking from a function that returns `anyhow::Result` is a great pattern. Is there a reason to not just return `Err(whatever_msg)`?

Reply:
I wanted that test case to fail and the CI running it to fail too in case the test didn't run as expected. This was the most intuitive way for me to achieve this actually 😅 

Is that the same for returning `Err(msg)`?

Reply:
It depends on how `anyhow::Result` is handled by the function caller, but usually it ends up panicking one way or the other. I guess this is opinionated so feel free to leave your option if you think it conveys your point in a more clear way.